### PR TITLE
Add instrumentation to teacher saving

### PR DESCRIPTION
**Pull request by megcrenshaw**
Adds an event every time a user submits or saves a teacher application.

## Links

- jira ticket: [PLAT-1473](https://codedotorg.atlassian.net/browse/PLAT-1473)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by megcrenshaw** [comment]
Not sure the best way to test this. Do I simulate a "save" and "submit" in `TeacherApplicationTest.js` (which requires knowing the structure of the `FormController` component)? Maybe I export functions in the FormControllerTest to simulate a save and submit? Open to suggestions

**Comment by davidsbailey** [comment]
> Not sure the best way to test this. Do I simulate a "save" and "submit" in `TeacherApplicationTest.js` (which requires knowing the structure of the `FormController` component)? Maybe I export functions in the FormControllerTest to simulate a save and submit? Open to suggestions

Good question Meg, let's discuss at stand up.

**Review by davidsbailey — APPROVED** [review]

File: apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx [hunk 1]
```diff
@@ -88,6 +87,9 @@ const TeacherApplication = props => {
     if (nominated) {
       url += '?nominated=true';
     }
+    if (savedStatus === 'incomplete') {
+      url += '?returned=true';
+    }
```
Comment (tim-dot-org) [inline apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: might want to upgrade this to use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams or something similar or else you'd potentially end up with ?nominated=true?returned=true which isn't a valid url

Comment (tim-dot-org) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: oh the query-string library already being used there will take care of it too

Comment (megcrenshaw) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: Ahh good to know. What *should* the url look like if it's both nominated and savedStatus? I was thinking I don't want to change the nominated one if possible

Comment (megcrenshaw) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: Or should this be different? Seems like the default is the `&`: `foo=bar&abc=xyz&abc=123` (https://nodejs.org/api/querystring.html#querystringparsestr-sep-eq-options)

Comment (tim-dot-org) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: it'd get an `&` between each query param, so `?nominated=true&returned=true` or `?returned=true` or `?returned=true&nominated=true`

Comment (megcrenshaw) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: Not sure who to contact about Google Analytics if we change it ...

Comment (megcrenshaw) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: Okay, I think that's easy for GA people to change ... will update!

Comment (tim-dot-org) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: shouldn't be anything that needs to change on the GA side! this is a standardized way of constructing urls

Comment (megcrenshaw) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:96]: Ah, cool beans. Just updated it! Might should add test coverage for this now ... let me know what you think, and I'll check in the morning.

**Review by tim-dot-org — APPROVED** [review]
looks good!

**Pull request merged**